webknossos/webknossos/annotation/annotation.py

```diff
@@ -107,9 +117,28 @@ def save_volume_annotation(
         """
 
         # todo pylint: disable=fixme
-        assert self._nml.volume is not None
-        volume_zip_path = self._nml.volume.location
-        assert volume_zip_path in self._filelist
+        assert len(self._nml.volumes) > 0
+
+        if len(self._nml.volumes) == 1:
+            volume_zip_path = self._nml.volumes[0].location
+        else:
+            assert source_volume_name != None, (
+                "The annotation contains multiple volume layers. "
+                "Please specify which layer should be used via `source_volume_name`."
+            )
+
+            volume = next(
+                filter(
+                    lambda volume: volume._get_name_or_id() == source_volume_name,
+                    self._nml.volumes,
+                )
+            )
```

Comment:
Please also catch the case where nothing matches (I guess a `StopIteration` is reaised then)

Reply:
Alternatively, you can also initialize `volume_zip_path` and use a regular for loop with an if and a `break`, might even be more readable, but no strong opinion :shrug:

Reply:
good idea. i went for the for-loop and added an assertion so that the error message looks nicer than the stopiteration.